hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java

```diff
@@ -549,6 +564,8 @@ protected void initializeOmTables(boolean addCacheMetrics)
     deletedTable = this.store.getTable(DELETED_TABLE, String.class,
         RepeatedOmKeyInfo.class);
     checkTableStatus(deletedTable, DELETED_TABLE, addCacheMetrics);
+    // Currently, deletedTable is the only table that will need the table lock
```

Comment:
Don't we need a table lock for deletedDirTable as well ?

Reply:
No. `deletedDirTable` entries "trickle down" to `deletedTable` eventually. See `DirectoryDeletingService`

Reply:
This reminds me that we at least need to hold the write lock in `OMDirectoriesPurgeResponseWithFSO` as well.

Reply:
It looks to me the parameter for the `deletedTable` put operation in `OMDirectoriesPurgeResponseWithFSO` could be wrong. It does not seem to be considering the case where `deletedKey` could have already existed in the `deletedTable`:

https://github.com/apache/ozone/blob/5eead92666b374c4bb912f332488acb23f9a81df/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/key/OMDirectoriesPurgeResponseWithFSO.java#L114-L122

The correct action would be to append the key to the existing `RepeatedOmKeyInfo` value in the `deletedTable` in this case.


Regarding the lock, I will put a TODO here at the moment. We will need to revisit the locking part in HDDS-8067.

Reply:
yup